From c24fe0634f1f4f730ded955c69b20f1fc8b0a2d5 Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Thu, 14 Nov 2024 12:37:11 -0700
Subject: [PATCH 1/2] Fix a segfault in wlEglCreatePlatformWindowSurfaceHook

In the error cleanup path in wlEglCreatePlatformWindowSurfaceHook, don't
try to dereference the WlEglSurface if we never allocated it.
---
 src/wayland-eglsurface.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/wayland-eglsurface.c b/src/wayland-eglsurface.c
index ae6cafc..16161f0 100644
--- a/src/wayland-eglsurface.c
+++ b/src/wayland-eglsurface.c
@@ -2843,15 +2843,14 @@ EGLSurface wlEglCreatePlatformWindowSurfaceHook(EGLDisplay dpy,
     return surface;
 
 fail:
-    if (surface->drmSyncobjHandle) {
-        drmSyncobjDestroy(display->drmFd, surface->drmSyncobjHandle);
-    }
-
     if (drmSyncobjFd > 0) {
         close(drmSyncobjFd);
     }
 
     if (surface) {
+        if (surface->drmSyncobjHandle) {
+            drmSyncobjDestroy(display->drmFd, surface->drmSyncobjHandle);
+        }
         wlEglDestroySurface(display, surface);
     }
 
-- 
2.43.0

